backend/bitboxbase/bitboxbase.go

```diff
@@ -228,44 +281,318 @@ func (base *BitBoxBase) RPCClient() *rpcclient.RPCClient {
 // MiddlewareInfo returns the received MiddlewareInfo packet from the rpcClient
 func (base *BitBoxBase) MiddlewareInfo() (rpcmessages.SampleInfoResponse, error) {
 	if !base.active {
-		err := errp.New("Attempted a call to non-active base")
-		return rpcmessages.SampleInfoResponse{}, err
+		return rpcmessages.SampleInfoResponse{}, errp.New("Attempted a call to non-active base")
 	}
 	return base.rpcClient.GetSampleInfo()
 }
 
 // VerificationProgress returns the received VerificationProgress packet from the rpcClient
 func (base *BitBoxBase) VerificationProgress() (rpcmessages.VerificationProgressResponse, error) {
 	if !base.active {
-		err := errp.New("Attempted a call to non-active base")
-		return rpcmessages.VerificationProgressResponse{}, err
+		return rpcmessages.VerificationProgressResponse{}, errp.New("Attempted a call to non-active base")
 	}
 	return base.rpcClient.GetVerificationProgress()
 }
 
-// SyncWithOption returns true if the chosen sync option was executed successfully
-func (base *BitBoxBase) SyncWithOption(option SyncOption) (bool, error) {
+// ReindexBitcoin returns true if the chosen sync option was executed successfully
+func (base *BitBoxBase) ReindexBitcoin() error {
+	if !base.active {
+		return errp.New("Attempted a call to non-active base")
+	}
+	base.log.Println("bitboxbase is making a ReindexBitcoin call")
+	reply, err := base.rpcClient.ReindexBitcoin()
+	base.changeStatus(bitboxbasestatus.StatusInitialized)
+	if err != nil {
+		return err
+	}
+	if !reply.Success {
+		return &reply
+	}
+	return nil
+}
+
+// ResyncBitcoin returns true if the chosen sync option was executed successfully
+func (base *BitBoxBase) ResyncBitcoin() error {
+	if !base.active {
+		return errp.New("Attempted a call to non-active base")
+	}
+	base.log.Println("bitboxbase is making a ResyncBitcoin call")
+	reply, err := base.rpcClient.ResyncBitcoin()
+	base.changeStatus(bitboxbasestatus.StatusInitialized)
```

Comment:
should this happen always or only if there was no error?

Reply:
See https://github.com/digitalbitbox/bitbox-wallet-app/pull/576#discussion_r330003725